Lib/zipfile/__init__.py

```diff
@@ -309,7 +309,7 @@ def _EndRecData(fpin):
         fpin.seek(-sizeEndCentDir, 2)
     except OSError:
         return None
-    data = fpin.read()
+    data = fpin.read(sizeEndCentDir + 1)
```

Comment:
We've just seeked back `sizeEndCentDir` bytes so should be ok reading the same number of bytes

```suggestion
    data = fpin.read(sizeEndCentDir)
```

Reply:
There is a slight behavior change, which may entirely be okay, if don't do the `+1` compared to the `read()` without a size (`readall`). The extra byte is needed so `read()` does a second `os.read()` to try getting that byte and either gets more data than expected (someone wrote to the file while / changed unexpectedly), or the second `os.read()` returns size 0 which indicates "this is EOF" (how `readall` decides to stop). The extra byte case it felt like was trying to be tested for just below in the `len(data) == sizeEndCentDir` case.

Happy to defer to others judgment around what zipfile should do in this case, it's reasonable to me to not read to EOF.

Reply:
`_Py_read` not `os.read` actually (os.read is a whole different codepath)

Reply:
I don't think it is required but I'm also happy to defer to others.

Here's the issue / PR that added the length check for context:
https://bugs.python.org/issue4844
https://github.com/python/cpython/commit/d2b1527f1418f4bac9d05933623ef1a9d1948f61

That check may have been added along with other similar checks across this file but not actually necessary due to the try except block that precedes it.

Reply:
If we do end up keeping the `+ 1`, we should add a comment detailing why it is there.